lib/graphql/schema/warden.rb

```diff
@@ -106,6 +106,12 @@ def get_field(parent_type, field_name)
         @visible_parent_fields[parent_type][field_name]
       end
 
+      # @return [GraphQL::Argument, nil] The argument named `argument_name` on `parent_type`, if it exists and is visible
+      def get_argument(parent_type, argument_name)
+        argument = parent_type.arguments[argument_name]
```

Comment:
`#arguments` climbs the inheritance chain and merges argument definitions into a hash. When you only need one object, better to use `.get_argument` which searches the chain until it finds a hit, and doesn't merge any hashes: 

```suggestion
        argument = parent_type.get_argument(argument_name)
```

Reply:
👀  I may be missing something, but I don't see `get_argument` defined on `GraphQL::Schema::InputObject`

https://graphql-ruby.org/api-doc/1.10.11/GraphQL/Schema/InputObject.html

Reply:
Ah, sorry, I was thinking it would be around here, but I was mistaken: https://github.com/rmosolgo/graphql-ruby/blob/12947f427d7e4b3e590b743b3c2c3ef04ece2712/lib/graphql/schema/member/has_arguments.rb#L50-L59

It could be added like get_field: 

https://github.com/rmosolgo/graphql-ruby/blob/12947f427d7e4b3e590b743b3c2c3ef04ece2712/lib/graphql/schema/member/has_fields.rb#L29-L40

Reply:
Good idea! Added :)